config/terraform_cloud_workspace_test.go

```diff
@@ -0,0 +1,316 @@
+		},
+		{
+			"nil_both",
+			nil,
+			nil,
+			nil,
+		},
+		{
+			"empty",
+			&TerraformCloudWorkspaceConfig{},
+			&TerraformCloudWorkspaceConfig{},
+			&TerraformCloudWorkspaceConfig{},
+		},
+		{
+			"execution_mode_overrides",
+			&TerraformCloudWorkspaceConfig{ExecutionMode: String("remote")},
+			&TerraformCloudWorkspaceConfig{ExecutionMode: String("agent")},
+			&TerraformCloudWorkspaceConfig{ExecutionMode: String("agent")},
+		},
+		{
+			"execution_mode_empty_a",
+			&TerraformCloudWorkspaceConfig{},
+			&TerraformCloudWorkspaceConfig{ExecutionMode: String("agent")},
+			&TerraformCloudWorkspaceConfig{ExecutionMode: String("agent")},
+		},
+		{
+			"execution_mode_empty_b",
+			&TerraformCloudWorkspaceConfig{ExecutionMode: String("agent")},
+			&TerraformCloudWorkspaceConfig{},
+			&TerraformCloudWorkspaceConfig{ExecutionMode: String("agent")},
+		},
+		{
+			"agent_pool_id_overrides",
+			&TerraformCloudWorkspaceConfig{AgentPoolID: String("apool-1")},
+			&TerraformCloudWorkspaceConfig{AgentPoolID: String("apool-2")},
+			&TerraformCloudWorkspaceConfig{AgentPoolID: String("apool-2")},
+		},
+		{
+			"agent_pool_id_empty_a",
+			&TerraformCloudWorkspaceConfig{},
+			&TerraformCloudWorkspaceConfig{AgentPoolID: String("apool-2")},
+			&TerraformCloudWorkspaceConfig{AgentPoolID: String("apool-2")},
+		},
+		{
+			"agent_pool_id_empty_b",
+			&TerraformCloudWorkspaceConfig{AgentPoolID: String("apool-1")},
+			&TerraformCloudWorkspaceConfig{},
+			&TerraformCloudWorkspaceConfig{AgentPoolID: String("apool-1")},
+		},
+		{
+			"agent_pool_name_overrides",
+			&TerraformCloudWorkspaceConfig{AgentPoolName: String("agent_pool_a")},
+			&TerraformCloudWorkspaceConfig{AgentPoolName: String("agent_pool_b")},
+			&TerraformCloudWorkspaceConfig{AgentPoolName: String("agent_pool_b")},
+		},
+		{
+			"agent_pool_name_empty_a",
+			&TerraformCloudWorkspaceConfig{},
+			&TerraformCloudWorkspaceConfig{AgentPoolName: String("agent_pool_b")},
+			&TerraformCloudWorkspaceConfig{AgentPoolName: String("agent_pool_b")},
+		},
+		{
+			"agent_pool_name_empty_b",
+			&TerraformCloudWorkspaceConfig{AgentPoolName: String("agent_pool_a")},
+			&TerraformCloudWorkspaceConfig{},
+			&TerraformCloudWorkspaceConfig{AgentPoolName: String("agent_pool_a")},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			r := tc.a.Merge(tc.b)
+			assert.Equal(t, tc.r, r)
+		})
+	}
+}
+
+func TestTerraformCloudWorkspaceConfig_Finalize(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
+		name string
+		c    *TerraformCloudWorkspaceConfig
+		r    *TerraformCloudWorkspaceConfig
+	}{
+		{
+			"nil",
+			nil,
+			nil,
+		},
+		{
+			"empty",
+			&TerraformCloudWorkspaceConfig{},
+			&TerraformCloudWorkspaceConfig{
+				ExecutionMode: String("remote"),
+				AgentPoolID:   String(""),
+				AgentPoolName: String(""),
+			},
+		},
+		{
+			"fully_configured",
+			&TerraformCloudWorkspaceConfig{
+				ExecutionMode: String("remote"),
```

Comment:
since "remote" is the default value that's being set by the function, maybe a better test would be to use a different string (i.e. agent) here. Just for future proofing

Reply:
I changed from not setting remote as default, but that's still a valid point in case we change this in the future!